src/types/scheduling.ts

```diff
@@ -197,6 +197,8 @@ export type SchedulingGoalPlanSpecification = {
   specification_id: number;
 };
 
+export type SchedulingGoalPlanSpecificationUpdate = SchedulingGoalPlanSpecification & { files: File[] | null };
```

Comment:
you can maybe make `files` optional via `files?: File[] | null` so that you aren't forced to provide an empty array of files for unrelated updates.